### PR TITLE
Feature/list view of users

### PR DESCRIPTION
**Pull request by hammad-Ikhlaq-7vals**
Now upon click the user in list you move towards show

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
+              <a href="<%= 'users/' + user.sequence_num.to_s %>" class="list-group-item list-group-item-action">
+                <span class="col-md-3">
+                    <%= user.name %>
+                </span>
+                <span class="col-md-3">
+                  <span class="badge badge-primary badge-pill float-right">
+                    <%= user.email %>
+                  </span>
+                </span>
+              </a>
```
Comment (hafsabatool7) [inline app/views/users/index.html.erb]: Use link_to

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -9,41 +9,43 @@
...
+          </div>
+      </div>
+      <div class="row">
+          <div class="col-md-6">
+              <label><%= t('users.company').capitalize() %>:</label>
+          </div>
+          <div class="col-md-6">
+              <p><%= @user.company.name %></p>
+          </div>
+      </div>
```
Comment (hafsabatool7) [inline app/views/users/show.html.erb]: Do we need to display company?

File: app/views/users/index.html.erb (hunk 1 above)
Comment (hammad-Ikhlaq-7vals) [reply app/views/users/index.html.erb]: done

File: app/views/users/show.html.erb (hunk 2 above)
Comment (hammad-Ikhlaq-7vals) [reply app/views/users/show.html.erb]: Yes you are right. He already knows that he has login into it.

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -34,6 +34,22 @@ def update
       end
   end
 
+  def filters
+    if(params[:search].present?)
+      if params[:search].eql? User::ROLES[0]
+        @users = @users.where(role: 'STAFF')
```
Comment (hafsabatool7) [inline app/controllers/users_controller.rb]: Use constant. Also on line 42 and 44

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -34,6 +34,22 @@ def update
...
+      elsif params[:search].eql? User::ROLES[1]
+        @users = @users.where(role: 'ADMIN')
+      elsif params[:search].eql? User::ROLES[2]
+        @users = @users.where(role: 'OWNER')
+      end
+    end
+    @pagy, @users = pagy(@users.order(created_at: :desc), items: 5)
```
Comment (hafsabatool7) [inline app/controllers/users_controller.rb]: Use page size constant

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -34,6 +34,22 @@ def update
       end
   end
 
+  def filters
```
Comment (hafsabatool7) [inline app/controllers/users_controller.rb:37]: Add ability for staff based on your usecase

File: app/controllers/users_controller.rb (hunk 3 above)
Comment (hammad-Ikhlaq-7vals) [reply app/controllers/users_controller.rb]: done

File: app/controllers/users_controller.rb (hunk 4 above)
Comment (hammad-Ikhlaq-7vals) [reply app/controllers/users_controller.rb]: done

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -10,7 +10,7 @@ def show
   end
 
   def index
-    @pagy, @users = pagy(@users,  items: Company::PAGE_SIZE)
+    @pagy, @users = pagy(@users,  items: User::PAGE_SIZE)
```
Comment (hafsabatool7) [inline app/controllers/users_controller.rb]: The PAGE_SIZE defined in the company model is the standard for all resources. You don't need to declare a new one. Just use that everywhere

File: app/models/ability.rb [hunk 7]
```diff
@@ -24,6 +24,7 @@ def initialize(user)
 
     can :read, User, sequence_num: user.sequence_num, company_id: user.company_id
```
Comment (hafsabatool7) [inline app/models/ability.rb]: Remove now that you have added new ability for read

File: config/locales/en.yml [hunk 8]
```diff
@@ -153,6 +153,13 @@ en:
     company: 'company'
     all_users: 'All Users'
     no_user_to_show: 'No User to show...'
+    user_created_successfully: 'User created successfully'
```
Comment (hafsabatool7) [inline config/locales/en.yml:155]: Can we not use the generic key defined on line 249?

File: app/controllers/users_controller.rb (hunk 5 above)
Comment (hammad-Ikhlaq-7vals) [reply app/controllers/users_controller.rb:37]: done

File: app/controllers/users_controller.rb (hunk 6 above)
Comment (hammad-Ikhlaq-7vals) [reply app/controllers/users_controller.rb]: done

File: app/models/ability.rb (hunk 7 above)
Comment (hammad-Ikhlaq-7vals) [reply app/models/ability.rb]: done

File: config/locales/en.yml (hunk 8 above)
Comment (hammad-Ikhlaq-7vals) [reply config/locales/en.yml:155]: done

File: config/locales/en.yml [hunk 9]
```diff
@@ -69,6 +69,7 @@ en:
     type_field: 'Type'
     created_field: 'Created'
     actions_field: 'Actions'
+    role_field: 'Role'
```
Comment (hafsabatool7) [inline config/locales/en.yml]: Use existing key

Comment (hammad-Ikhlaq-7vals) [reply config/locales/en.yml]: done

**Pull request merged**